opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.hpp

```diff
@@ -113,6 +117,11 @@ class WellTestState {
 private:
     std::vector<ClosedWell> wells;
     std::vector<ClosedCompletion> completions;
+
+
+    ClosedWell& getWell(const std::string& well_name, WellTestConfig::Reason reason);
+
+    void reCloseWell(const std::string& well_name, WellTestConfig::Reason reason, const double sim_time);
```

Comment:
Is it really necessary to have both `closeWell()` and `reCloseWell()` methods; I would assume everything could be in the `closeWell()`method - which would first check if the well is already in the list. The `getWell()` method could be private - and return a pointer:

```C++
void closeWell() {
      auto * well_ptr = this->getWell();
      if (!well_ptr) {
           this->wells.push_back({});
           well_ptr = std::addressof( this->wells.back() );
       }
       well_ptr->closed = true;
       ...
```

Reply:
Sure.  I think your suggestion about `getWell()` is very good, which is something I am looking for.